desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -89,23 +89,26 @@ def handle_on_message(channel_id, bot_id, text, user_id):
       if not response['ok']:
         raise PopupException(_("Error posting message"), detail=response["error"])
 
-
 def handle_on_link_shared(channel_id, message_ts, links):
   for item in links:
     path = urlsplit(item['url'])[2]
     id_type, qid_or_uuid = urlsplit(item['url'])[3].split('=')
 
-    if path == '/hue/editor' and id_type == 'editor':
-      doc = Document2.objects.get(id=qid_or_uuid)
-    elif path == '/hue/gist' and id_type == 'uuid' and ENABLE_GIST_PREVIEW.get():
-      doc = _get_gist_document(uuid=qid_or_uuid)
-    else:
-      raise PopupException(_("Cannot unfurl link"))
-
-    doc_data = json.loads(doc.data)
-    statement = doc_data['snippets'][0]['statement_raw'] if id_type == 'editor' else doc_data['statement_raw']
-    dialect = doc_data['dialect'].capitalize() if id_type == 'editor' else doc.extra.capitalize()
-    created_by = doc.owner.get_full_name() or doc.owner.username
+    try:
+      if path == '/hue/editor' and id_type == 'editor':
+        doc = Document2.objects.get(id=qid_or_uuid)
+      elif path == '/hue/gist' and id_type == 'uuid' and ENABLE_GIST_PREVIEW.get():
+        doc = _get_gist_document(uuid=qid_or_uuid)
+      else:
+        raise PopupException(_("Cannot unfurl link"))
+
+      doc_data = json.loads(doc.data)
+      statement = doc_data['snippets'][0]['statement_raw'] if id_type == 'editor' else doc_data['statement_raw']
+      dialect = doc_data['dialect'].capitalize() if id_type == 'editor' else doc.extra.capitalize()
+      created_by = doc.owner.get_full_name() or doc.owner.username
```

Comment:
Document2.DoesNotExist: can't happen here, could you move up the except to below line 103?

(to avoid big global try/except)

Reply:
Fixed